src/module/rules/rule-element/item-alteration/alteration.ts

```diff
@@ -389,6 +392,24 @@ class ItemAlteration extends foundry.abstract.DataModel<RuleElementPF2e, ItemAlt
                 }
                 return;
             }
+            case "range-increment": {
+                const validator = ITEM_ALTERATION_VALIDATORS[this.property];
+                if (!validator.isValid(data)) return;
+                if (!(data.item instanceof WeaponPF2e) || !data.item.system.range) return;
```

Comment:
Why do you need to check if its a weapon? It appears the other alterations don't need to perform this check and cast appropriately after running isValid().